### ISSUE TITLE
Add an `--dump-empty-db` parameter to control whether to dump empty databases

### ISSUE THREAD

**Issue by cyliu0** [pingcap/dumpling#96]
```
mysql root@127.0.0.1:test> create database cy
Query OK, 0 rows affected
Time: 0.004s
mysql root@127.0.0.1:test> create table cy.t (a int)
Query OK, 0 rows affected
Time: 0.006s

bash> ./bin/dumpling -f "*.cy"
bash> ls export-2020-06-08T17:40:02+08:00
cy-schema-create.sql   metadata               test-schema-create.sql
```

It should not dump the schemas cy and test.

**Comment by kennytm** [comment]
This is intended, because upstream may create the tables `cy.cy` / `test.cy`.

**Comment by cyliu0** [comment]
According to the description `-f, --filter stringArray      filter to select which tables to dump (default [*.*])`, I think it should dump the matched tables and the databases which the matched tables located

**Comment by kennytm** [comment]
Keeping the `*-schema-create.sql` file for empty database is exactly the behavior of mydumper. Try:

```sh
echo 'cy.t' > omit.txt
mydumper -O omit.txt -h 127.0.0.1 -P 4000 -o aaaaa
ls aaaaa/cy*
```

Also, in mydumper if you use `-x '\.cy$'` it will never create the `*-schema-create.sql` even if a `test.cy` table exists. I don't think we should copy mydumper's behavior here.

**Comment by lichunzhu** [comment]
I agree with kennytm. If we really want to get rid of empty databases may we should add another argument.

**Comment by cyliu0** [comment]
> Keeping the `*-schema-create.sql` file for empty database is exactly the behavior of mydumper. Try:
> 
> ```shell
> echo 'cy.t' > omit.txt
> mydumper -O omit.txt -h 127.0.0.1 -P 4000 -o aaaaa
> ls aaaaa/cy*
> ```
> 
> Also, in mydumper if you use `-x '\.cy$'` it will never create the `*-schema-create.sql` even if a `test.cy` table exists. I don't think we should copy mydumper's regex behavior here.

Yes. I do believe that we need to create `*-schema-create.sql` when we found the matched tables (or databases?). But in this case, it will create empty databases even if there is nothing matched.  Are you sure about this behavior is expected?

**Comment by kennytm** [comment]
Yes it is expected. We can add a `--dump-empty-db=false` parameter, but the current behavior is by design.

**Comment by kennytm** [comment]
the `--dump-empty-database` parameter has been implemented in #103, and we are not going to change the behavior of `--filter`, thus closing.
